mantle/kola/tests/ignition/units.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ignition
+
+import (
+	"strings"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform/conf"
+)
+
+func init() {
+	register.RegisterTest(&register.Test{
+		Name:        "coreos.ignition.instantiated.enable-service",
+		Run:         enableSystemdInstantiatedService,
+		ClusterSize: 1,
+		UserDataV3: conf.Ignition(`{
+    "ignition": {"version": "3.0.0"},
+    "systemd": {
+        "units": [{
+			"name": "echo@.service",
+			"contents": "[Unit]\nDescription=f\n[Service]\nType=oneshot\nExecStart=/bin/echo %i\nRemainAfterExit=yes\n[Install]\nWantedBy=multi-user.target\n"
+		  },
+		{
+		  "enabled": true,
+		  "name": "echo@bar.service"
+		},
+		{
+		  "enabled": true,
+		  "name": "echo@foo.service"
+		}]
+    }
+}`),
+		// Enabling systemd instantiated services doesn't support
+		// in a given system if the version of systemd is older than
+		// 240. RHCOS deosn't support this feature currently because
+		// its running an older version (v239) of systemd.
+		ExcludeDistros: []string{"rhcos"},
+	})
+}
+
+func enableSystemdInstantiatedService(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	fooStatus := c.MustSSH(m, "systemctl status echo@foo.service")
+	barStatus := c.MustSSH(m, "systemctl status echo@bar.service")
+	if strings.Contains(string(fooStatus), "inactive") && strings.Contains(string(barStatus), "inactive") {
+		c.Fatalf("services are not enabled or systemd-presets did not run")
+	}
```

Comment:
we should probably check them individually rather than together, right? i.e. if either is not enabled or not active then the test fails.

Reply:
I was thinking about failing the entire test if either of them isn't enabled/active rather than checking it individually.

Reply:
right what you said is what we want to do and also what I'm saying we should do. What I'm saying here is I don't think your code is doing that. The `if () && ()` that you are doing I think will only fail this test if **both** are inactive. What if one is active and one isn't? Then the current code would pass.

Reply:
> right what you said is what we want to do and also what I'm saying we should do. What I'm saying here is I don't think your code is doing that. The `if () && ()` that you are doing I think will only fail this test if **both** are inactive. What if one is active and one isn't? Then the current code would pass.

Yes, you're right. I have written that condition taking the assumption that those two services will either be active or inactive. I wasn't sure (specifically for instantiated services) if we need to consider a case where only one of them will be active. Thanks for the clarification.